cmake/SuperBuild.cmake

```
@@ -49,7 +49,7 @@ if(UAGENT_P2P_PROFILE)
 endif()
 
 # Fast CDR.
-if(NOT UAGENT_USE_SYSTEM_FASTDDS)
+if(NOT UAGENT_USE_SYSTEM_FASTCDR)
```

```suggestion
if(UAGENT_FAST_PROFILE AND NOT UAGENT_USE_SYSTEM_FASTCDR)
```


Is FastCDR part of the FastMiddleware profile? Check:
https://github.com/eProsima/Micro-XRCE-DDS-Agent/blob/3fd419f943cacb35d792fd4558b978a1f57bb3ee/cmake/SuperBuild.cmake#L83-L86

---

No, the XRCE-DDS Agent uses FastCDR for message serialization in any case. For example `UAGENT_CED_PROFILE` uses also FastCDR but not FastDDS, so FastCDR is required even if `UAGENT_FAST_PROFILE` is not enabled.

---

Ok then